implementation/domain/web/jbb-board-web/src/main/java/org/jbb/board/web/forum/controller/AcpForumController.java

```diff
@@ -0,0 +1,166 @@
+import org.jbb.board.web.forum.form.ForumForm;
+import org.jbb.lib.mvc.SimpleErrorsBindingMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.servlet.mvc.support.RedirectAttributes;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import lombok.extern.slf4j.Slf4j;
+
+@Slf4j
+@Controller
+@RequestMapping("/acp/general/forums/forum")
+public class AcpForumController {
+    private static final String VIEW_NAME = "acp/general/forum";
+    private static final String REDIRECT_TO_FORUM_MANAGEMENT = "redirect:/acp/general/forums";
+
+    private static final String FORUM_FORM = "forumForm";
+    private static final String FORUM_DELETE_FORM = "forumCategoryDeleteForm";
+    private static final String FORUM_ROW = "forum";
+
+    private final BoardService boardService;
+    private final ForumService forumService;
+    private final ForumCategoryService forumCategoryService;
+    private final SimpleErrorsBindingMapper errorMapper;
+
+    @Autowired
+    public AcpForumController(BoardService boardService, ForumService forumService,
+                              ForumCategoryService forumCategoryService, SimpleErrorsBindingMapper errorMapper) {
+        this.boardService = boardService;
+        this.forumService = forumService;
+        this.forumCategoryService = forumCategoryService;
+        this.errorMapper = errorMapper;
+    }
+
+    @RequestMapping(method = RequestMethod.GET)
+    public String forumGet(@RequestParam(value = "id", required = false) Long forumId, Model model) {
+        ForumForm form = new ForumForm();
+
+        List<ForumCategory> allCategories = boardService.getForumCategories();
+        List<ForumCategoryRow> categoryDtos = allCategories.stream()
+                .map(category -> mapToForumCategoryDto(category))
+                .collect(Collectors.toList());
+        model.addAttribute("availableCategories", categoryDtos);
+
+        if (forumId != null) {
+            Forum forum = forumService.getForum(forumId);
+            form.setId(forum.getId());
+            form.setName(forum.getName());
+            form.setDescription(forum.getDescription());
+            form.setLocked(forum.isLocked());
+
+            ForumCategory category = forumCategoryService.getCategoryWithForum(forum);
+            form.setCategoryId(category.getId());
+        } else {
+            form.setCategoryId(Iterables.getFirst(allCategories, null).getId());
+        }
+
+        if (!model.containsAttribute(FORUM_FORM)) {
```

Comment:
FORUM_FORM can be present in model when redirection after validation error happended (see `catch` block in `forumPost` method). Without this if-check form will be cleared every time, but after validation error we want to remember provided data